secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/SecretManagerProducer.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkiverse.googlecloudservices.secretmanager.runtime;
+
+import java.io.IOException;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceSettings;
+
+import io.quarkiverse.googlecloudservices.common.GcpConfiguration;
+
+/**
+ * Producer for the Google Cloud Secret Manager service.
+ */
+@ApplicationScoped
+public class SecretManagerProducer {
+
+    @Inject
+    GoogleCredentials googleCredentials;
+
+    @Inject
+    GcpConfiguration gcpConfiguration;
+
+    @Produces
+    @Singleton
+    @Default
+    public SecretManagerServiceClient secretManagerClient() throws IOException {
```

Comment:
Glad to see this 😀

I was going to ask for a way to inject it on the original PR and for more auth capabilities but as you add it in this PR it's not needed on the other one.